packages/block-editor/src/components/block-popover/index.js

```diff
@@ -88,6 +98,29 @@ function BlockPopover(
 
 		return {
 			getBoundingClientRect() {
+				// The zoom out view has a vertical block toolbar that should always
+				// be on the edge of the canvas. This condition changes the anchor
+				// of the toolbar to the section instead of the block to handle blocks
+				// that are not full width.
+				if ( isZoomOut ) {
+					const selectedBlockRect =
+						getVisibleElementBounds( selectedElement );
+					const sectionRootElementRect = getVisibleElementBounds(
+						selectedElement.parentElement
```

Comment:
Here we want the element of the parent block, in case the block is inside a container, is that right? Looking at `parentElement` is not reliable, it will work correctly only if the parent container has child blocks directly inside its main block element. If there are any "intermediate" elements, the alignment will be wrong.

The code should probably use something like `useBlockElement( parentClientId )`.

Reply:
Hmm, indeed. I guess I was fooled by the way selection currently works in zoom out where nested elements cannot be selected. But getting the element via parent **section** clientId should be more bullet proof once #65702 lands.

Reply:
> But getting the element via parent section `clientId` should be more bullet proof once #65702 lands.

I see there's a new `getParentSectionBlock` selector, and that's what you eventually want to use, is that right? Then when you have the section block ID, you use `useBlockElement` to retrieve its DOM element.

One thing to watch for is that `useBlockElement` initially returns `null`, because the elements become known only after some ref callbacks and effects run, they are not known during the initial render. That's why the `BlockPopover` render function has conditions like:
```js
if ( ! selectedElement || ( bottomClientId && ! lastSelectedElement ) ) {
  return null;
}
```
These conditions will need to be updated so that the component "waits" until also the `sectionRootElement` is ready.

Reply:
@jsnajdr our Api just grows by the day, good find with that parent selector, I was finding it manually in 76969f1bdc82607f024602688c01cb2511dc260a 🙇🏻 

Anyway code is updated now LMK how it looks to you (also you @getdave)